### PR TITLE
[change] Moved DeviceNode data migration to create_device_nodes management command #92

### LINKED ISSUE #92: [network-topology] Replace DeviceNode data migration with management command
When using the integration with openwisp-controller, instead of using a data migration to create the initial DeviceNode objects, we should use a management command, because the logic to create the initial DeviceNode objects is in a method which cannot be used during a migration because concrete models cannot be used during data migrations without risking future issues when adding/removing fields.

We'll also need to add a few lines in the readme regarding this command (when to use it and an example).

### PR DESCRIPTION
**Pull request by NoumbissiValere**
Closes #92

### CODE REVIEW

File: openwisp_network_topology/integrations/device/management/commands/__init__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    help = 'Create initial DeviceNode objects'
+
+    def handle(self, *args, **kwargs):
+        Node = swapper.load_model('topology', 'Node')
+        DeviceNode = swapper.load_model('topology_device', 'DeviceNode')
+        queryset = Node.objects.select_related('topology').filter(
+            topology__parser='netdiff.OpenvpnParser'
+        )
+        for node in queryset.iterator():
+            DeviceNode.auto_create(node)
```
Comment (nemesisdesign) [inline openwisp_network_topology/integrations/device/management/commands/__init__.py:15]: looks like a test is missing for this

File: README.rst [hunk 2]
```diff
@@ -294,6 +294,16 @@ by default the data is added to the first found organization, eg::
 
 **Note**: you can follow the `tutorial to migrate database from django-netjsongraph <https://github.com/openwisp/django-netjsongraph/blob/master/README.rst>`_.
 
+create_device_nodes
+^^^^^^^^^^^^^^^^^^^
```
Comment (nemesisdesign) [inline README.rst:298]: always look at how the documentation you write looks: https://github.com/openwisp/openwisp-network-topology/tree/issue-92-DeviceNode-management-command#upgrade_from_django_netjsongraph

See the difference?

Needs to be:

```
``create_device_nodes``
^^^^^^^^^^^^^^^^^^^^^^^
```

BTW, also look at this: https://github.com/openwisp/openwisp-network-topology#openwisp-network-topology-api-auth-required, I didn't spot it before.. please stay consistent with the rest of the codebase.

File: README.rst [hunk 3]
```diff
@@ -294,6 +294,16 @@ by default the data is added to the first found organization, eg::
...
+
+The ``create_device_nodes`` management command is used to create ``DeviceNode`` objects when
+this module is integrated with ``openwisp_controller``. It can be used as follows:
```
Comment (nemesisdesign) [inline README.rst:301]: I'd say something like:

> This management command can be used to create the initial ``DeviceNode`` relationships when the integration with openwisp-controller (add a link to #integration-with-openwisp-controller-and-openwisp-monitoring) is installed on a pre-existing systems which already has devices in it.

Then, I would add a link to this section at the bottom of #integration-with-openwisp-controller-and-openwisp-monitoring, something like:

> If you're enabling this integration on a pre-existing system, use the create_device_nodes management command (add a link to the management command section) to create the relationships between device and nodes.

**Note**: the part between parenthesis are indications, not text to copy verbatim, ok?

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Great @NoumbissiValere, see below a few requests for improvement.

File: README.rst (hunk 2 above)
Comment (NoumbissiValere) [reply README.rst:298]: @nemesisdesign you were the one who discouraged double ticks from headings in one PR I send. see https://github.com/openwisp/openwisp-network-topology/pull/84#discussion_r471816966

Comment (nemesisdesign) [reply README.rst:298]: Read again: 

> do not use if for non-code text please

`create_device_nodes` is code.

Comment (NoumbissiValere) [reply README.rst:298]: Ok thanks

**Review by nemesisdesign — APPROVED** [review]
Great, thanks!

**Pull request merged**